### PR TITLE
Use AtomicU64 instead RwLock for tracking snapshot size.

### PR DESCRIPTION
**Pull request by hicqu**
It's easy to write bug for tracking snapshot size with `RwLock`, especially for now rustc is not smart enough to deconstruct `LockGuard`s ASAP. For example, in function `Snap::save_cf_files` the lock won't be released until the file crc32 is calculated. If the file is large, it will take a long while so that maybe block `raftstore` thread.
This PR fix this issue by replace `RwLock` with `AtomicU64`.

### CODE REVIEW

**Comment by breeswish** [comment]
LGTM. BTW I think rust will never deconstruct LockGuard ASAP because you can use scopes or drop to manually deconstruct it if necessary?

File: src/raftstore/store/snap.rs [hunk 1]
```diff
@@ -875,8 +873,7 @@ impl Snapshot for Snap {
         for cf_file in &self.cf_files {
             delete_file_if_exist(&cf_file.tmp_path);
             if file_exists(&cf_file.path) {
-                let mut size_track = self.size_track.wl();
-                *size_track = size_track.saturating_sub(cf_file.size);
+                self.size_track.fetch_sub(cf_file.size, Ordering::SeqCst);
```
Comment (siddontang) [inline src/raftstore/store/snap.rs]: can fetch_sub overflow here?

Comment (hicqu) [reply src/raftstore/store/snap.rs]: It can happen because now register a snapshot multi times is not forbidden by `SnapManager`.

**Comment by hicqu** [comment]
PTAL @siddontang .

File: src/raftstore/store/snap.rs [hunk 2]
```diff
@@ -874,12 +872,10 @@ impl Snapshot for Snap {
...
         for cf_file in &self.cf_files {
             delete_file_if_exist(&cf_file.tmp_path);
-            if file_exists(&cf_file.path) {
-                let mut size_track = self.size_track.wl();
-                *size_track = size_track.saturating_sub(cf_file.size);
-            }
-            delete_file_if_exist(&cf_file.path);
             delete_file_if_exist(&cf_file.clone_path);
+            if delete_file_if_exist(&cf_file.path) {
+                self.size_track.fetch_sub(cf_file.size, Ordering::SeqCst);
```
Comment (siddontang) [inline src/raftstore/store/snap.rs]: If overflow, can we not remove the snapshot anymore?

I still worry about it.

Comment (hicqu) [reply src/raftstore/store/snap.rs]: There won't be any overflow. We increase `size_track` when cf files is renamed from tmp files, and decrease it after delete disk files success.

**Comment by siddontang** [comment]
@hicqu maybe we can do a scale benchmark, I guess this can reduce the QPS shake.

**Comment by hicqu** [comment]
Ok, I will test it.

**Pull request merged**